src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java

```diff
@@ -58,104 +62,60 @@ public Event findById(Long id) {
-            throw eventEntityNotFoundException(e.getMessage(), "id = " + id);
+            throw new EventNotFoundException("Failed to obtain event during `select` {id = " + id + "}, EventDao.findById", e);
         } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "id = " + id);
+            throw new CRUDException("Error during `select` {id = " + id + "}, EventDao.findById", e);
         }
     }
 
-    @Override
-    public Event findByTransactionId(Long transactionId) {
+    public List<Event> findAll(Map<String, ?> params) {
+        final String where = Stream
+            .of("account_id", "warehouse_id", "author_id", "name", "type", "date", "after", "before")
+            .filter(params::containsKey)
+            .map(x -> buildSqlFilterCondition(x, params.get(x)))
+            .collect(Collectors.joining("\n and "));
+        final String query = String.format("select * from events where %s",
+            where.isBlank() ? "TRUE" : where);
         try {
-            return jdbcTemplate.queryForObject(Queries.SQL_SELECT_EVENT_BY_TRANSACTION_ID, eventRowMapper, transactionId);
+            return jdbcTemplate.query(query, eventRowMapper);
         } catch (EmptyResultDataAccessException e) {
-            throw eventEntityNotFoundException(e.getMessage(), "id = " + transactionId);
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "id = " + transactionId);
-        }
-    }
-
-    @Override
-    public List<Event> findByAccountId(Long accountId) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENT_BY_ACCOUNT_ID, eventRowMapper, accountId);
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
-        }
-    }
-
-    @Override
-    public List<Event> findByWarehouseId(Long warehouseId) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_WAREHOUSE_ID, eventRowMapper, warehouseId);
+            throw new EventNotFoundException("Failed to obtain event during `select` {" + where + "}, EventDao.findAll", e);
         } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
+            throw new CRUDException("Error during `select` {" + where + "}, EventDao.findAll", e);
         }
     }
 
-    @Override
-    public List<Event> findByAuthorId(Long authorId) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_AUTHOR_ID, eventRowMapper, authorId);
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
+    private String buildSqlFilterCondition(String columnName, Object columnValue) {
+        if (columnValue instanceof Collection && columnName != "type") {
+            String values = "";
+            for (Object value : (Collection<Object>) columnValue) {
+                System.out.println(value);
```

Comment:
?

Reply:
With it we can select events with 2+ parameters with the same name. 
E.g.  Map<String, ?> params.put("warehouse_id", [123, 124, 125])

Reply:
I mean what the heck System.out.println(value) in your code?